wweval/R/eval_post_process.R

```diff
@@ -47,433 +47,476 @@ eval_post_process_ww <- function(config_index,
-  ww_diagnostics <- ww_fit_obj$sampler_diagnostics(format = "df")
-  save_object("ww_diagnostics", output_file_suffix)
-  ww_diagnostic_summary <- ww_fit_obj$diagnostic_summary()
-  save_object("ww_diagnostic_summary", output_file_suffix)
-  errors <- ww_fit_obj$error
-  save_object("errors", output_file_suffix)
-  metadata <- ww_fit_obj$metadata()
-  raw_flags <- get_diagnostic_flags(
-    ww_fit_obj,
-    metadata$num_chains,
-    metadata$iter_sampling
-  )
-  save_object("raw_flags", output_file_suffix)
-
-  # Save some plots of the posterior parameters---------------------
-  inf_feedback <- ww_raw_draws |>
-    tidybayes::spread_draws(!!str2lang("infection_feedback")) |>
-    dplyr::mutate(
-      draw = .data$`.draw`,
-    ) |>
-    dplyr::select("infection_feedback", "draw")
-
-  p_inf <- ggplot(
-    inf_feedback,
-    aes(x = infection_feedback)
-  ) +
-    geom_histogram()
-
-  eta_sd <- ww_raw_draws |>
-    tidybayes::spread_draws(!!str2lang("eta_sd")) |>
-    dplyr::mutate(
-      draw = .data$`.draw`,
-    ) |>
-    dplyr::select("eta_sd", "draw")
-
-  p_eta_sd <- ggplot(
-    eta_sd,
-    aes(x = eta_sd)
-  ) +
-    geom_histogram()
+  # Format input hosp data in format the eval pipeline expects
+  input_hosp_data_wweval <- input_hosp_data |>
+    dplyr:::rename(
+      "daily_hosp_admits" = "count",
+      "pop" = "total_pop"
+    )
 
-  ggsave(p_inf,
-    filename = file.path(
-      output_dir, scenario,
-      forecast_date, "ww", location,
-      "inf_feedback.png"
-    ),
-    create.dir = TRUE
-  )
-  ggsave(p_eta_sd,
-    filename = file.path(
-      output_dir, scenario,
-      forecast_date, "ww", location,
-      "eta_sd.png"
-    ),
-    create.dir = TRUE
+  input_ww_data_wweval <- tryCatch(
+    {
+      input_ww_data |>
+        dplyr::mutate(
+          "ww" = exp(.data$log_genome_copies_per_ml),
+          "lod_sewage" = exp(.data$log_lod)
+        ) |>
+        dplyr::rename(
+          "ww_pop" = "site_pop",
+          "below_LOD" = "below_lod"
+        )
+    },
+    error = function(e) {
+      # Return an empty tibble if this errors
+      message("Caught an error: ", e$message)
+      return(tibble::tibble())
+    }
   )
 
-  save_table(
-    data_to_save = eta_sd,
-    type_of_output = "eta_sd",
-    output_dir = output_dir,
-    scenario = scenario,
-    forecast_date = forecast_date,
-    model_type = "ww",
-    location = location
-  )
-  save_table(
-    data_to_save = inf_feedback,
-    type_of_output = "inf_feedback",
-    output_dir = output_dir,
-    scenario = scenario,
-    forecast_date = forecast_date,
-    model_type = "ww",
-    location = location
+  # Get table of wastewater data flags
+  ww_data_flags <- tryCatch(
+    {
+      get_ww_data_flags(
+        input_ww_data_wweval,
+        forecast_date
+      )
+    },
+    error = function(e) {
+      # Return an empty tibble if this errors
+      message("Caught an error: ", e$message)
+      return(tibble::tibble())
```

Comment:
What does `get_ww_data_flags` return if there are no flags? If it's an empty tibble, you probably want something else when `get_ww_data_flags` errors

Reply:
If there are no flags, it returns a table of flags with FALSE for each flag. What it needs to handle here is if there is no wastewater data, which it will currently error on https://github.com/CDCgov/wastewater-informed-covid-forecasting/blob/hot-fix-full-targets-eval/wweval/R/get_table_sufficient_ww.R